util/helpers.go

```diff
@@ -1,6 +1,40 @@
 package util
 
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
 // Pointer returns the pointer of any type
 func Pointer[T any](t T) *T {
 	return &t
 }
+
+// RenderObjectLabel renders a 63 charater long unique label
+func RenderObjectLabel(i types.UID) string {
+	return fmt.Sprintf("CLi-%s", strings.ReplaceAll(string(i), "-", ""))
+}
+
+// CreateLinodeAPIFilter converts variables to API filter string
+func CreateLinodeAPIFilter(label string, tags []string) string {
+	filter := map[string]string{}
+
+	if label != "" {
+		filter["label"] = label
+	}
+
+	if len(tags) != 0 {
+		filter["tags"] = strings.Join(tags, ",")
+	}
+
+	rawFilter, err := json.Marshal(filter)
+	if err != nil {
+		// This should never happen
+		panic(err.Error() + " Oh, snap... Earth has over, we can't parse map[string]string to JSON! I'm going to die ...")
```

Comment:
i get this won't happen but a panic in a util scares me...  can we just log and return an empty filter?

Reply:
An empty filter would break correctness of list operations of the API. So we have to return the error (which never happens) and deal with it everywhere. WDYT?

Reply:
if you call `CreateLinodeAPIFilter("", nil))` you'll get `{}`, we can't return that as a base case without making things worse in the subsequent calls?

Reply:
> if you call `CreateLinodeAPIFilter("", nil))` you'll get `{}`, we can't return that as a base case without making things worse in the subsequent calls?

@luthermonson I'm worries about this list call for example. https://github.com/linode/cluster-api-provider-linode/pull/62/files#diff-8daab1045cecf480419b0a31c2f8558e51197d31a7be910386ef4e420db0f7a4R253 If `CreateLinodeAPIFilter` returns an empty filter, this call would list all of the instances, and we and in this error: https://github.com/linode/cluster-api-provider-linode/pull/62/files#diff-8daab1045cecf480419b0a31c2f8558e51197d31a7be910386ef4e420db0f7a4R303. 

This call https://github.com/linode/cluster-api-provider-linode/pull/62/files would end in https://github.com/linode/cluster-api-provider-linode/pull/62/files#diff-371045646e070a553a5d00d8927bc43b4c0d67da0b77e054a6ddcf5123b838f4R52. We select the first item. 

So i'm glad to change the code, but we have to count with some refactoring and in the future we have to remember to this behavior all the time. Otherwise, we should iterate on the wrong set of API objects.

Anyway this is a common pattern in go, maybe i can rename the function to `CreateLinodeAPIFilterOrDie` or `MustCreateLinodeAPIFilter` to turn it super goish.